react-gbajs/src/react/index.js

```diff
@@ -25,8 +26,24 @@ const ReactGbaJs = ({ romBufferMemory, volume }) => {
     setVolume(gba, volume)
   }, [gba, volume])
 
+  function handleSave() {
+    setSaveState(gba.freeze());
+  }
+
+  function handleRestore() {
+    if (saveState) {
+      gba.setRom(romBufferMemory.buffer);
+      gba.defrost(cloneDeep(saveState));
+      gba.runStable();
+    }
```

Comment:
I think it would be possible to restore without fully resetting the gba. But fully resetting it also doesn't really do much harm, and it's nice to know it's in a known/good state before restoring. shrug.